### PR TITLE
Fix unit test errors due to new version of pydantic

### PR DESCRIPTION
**Pull request by eggmaster**
Pydantic updated to v2.x, so a bunch of pipeline unit tests started failing. This patch follows the migration recommendations from pydantic to get the tests passing again.

### CODE REVIEW

File: src/bugjira/config.py [hunk 1]
```diff
@@ -1,38 +1,35 @@
...
+
     URL: constr(strip_whitespace=True, min_length=1)
     token_auth: constr(strip_whitespace=True, min_length=1)
 
-    class Config:
-        extra = 'forbid'
-
 
 class ConfigDict(BaseModel):
+    model_config = ConfigDict(extra='forbid')
```
Comment (jguiditta) [inline src/bugjira/config.py]: This seems oddly recursive, I wonder if we should consider renaming this class (could be in a different patch) to make it less confusing?

**Review by jguiditta — DISMISSED** [review]
One comment to consider for a future patch, unless I am just being dense. Otherwise, looks good!

File: src/bugjira/config.py (hunk 1 above)
Comment (eggmaster) [reply src/bugjira/config.py]: oops! good catch.

**Review by jguiditta — APPROVED** [review]

**Review by jpichon — APPROVED** [review]

File: src/bugjira/config.py (hunk 1 above)
Comment (jpichon) [reply src/bugjira/config.py]: I was wondering as well... and I think that's why we have to "import pydantic" instead of just having a specific class import on the next line, like for the other imports? After I asked the question, I realised it would conflict with our custom class, but would agree otherwise on maybe renaming for clarity in a later patch, if that is possible.
